lib/did-io.js

```diff
@@ -379,6 +382,127 @@ api.promises = function(options) {
   return papi;
 };
 
+/**
+ * Register a DID document with Authorization.io.
+ * @param [options] the options to use:
+ *          [baseUrl] the target endpoint for the request
+ *          [didDocument] the DID document to register.
+ *          [privateKeyPem] the private key used to sign to DID document.
+ *          [strictSSL] enable/disable strictSSL.
+ * @param callback(err) called once the operation completes.
+ */
+api.registerDid = function(options, callback) {
+  var _ = api.use('_');
+  var async = api.use('async');
+  var jsigs = api.use('jsigs');
+  var request = api.use('request');
+  // TODO: add a mechanism to skip waiting when it's known the DID has
+  // been registered, so long as it doesn't cause interdependencies with
+  // database initialization and so on (a mechanism other than just not
+  // specifying the DID in the config?)
+
+  // TODO: may need to also support passing a function to generate the baseUrl
+  var baseUrl;
+  if('baseUrl' in options) {
+    if(typeof options.baseUrl !== 'string') {
+      throw new Error('options.baseUrl must be a string.');
+    }
+    baseUrl = options.baseUrl;
+  } else {
+    baseUrl = defaults.baseUrl;
+  }
+
+  var did = options.didDocument.id;
+  var privateKeyPem = options.privateKeyPem;
+  var strictSSL = true;
```

Comment:
This should be set to `defaults.strictSSL` as `baseUrl` is above (and then initialize that to `true` above where `defaults.baseUrl` is also initialized). That way it can be set generally.